allennlp/data/dataset_readers/nlvr.py

```diff
@@ -87,53 +97,67 @@ def _read(self, file_path: str):
+        structured_representations : ``List[List[List[JsonDict]]]``
+            A list of Json representations of all the worlds. See expected format in this class' docstring.
+        labels : ``List[str]`` (optional)
+            List of string representations of the labels (true or false) corresponding to the
+            ``structured_representations``. Not required while testing.
         """
         # pylint: disable=arguments-differ
-        world = NlvrWorld(structured_representation)
+        worlds = [NlvrWorld(data) for data in structured_representations]
         tokenized_sentence = self._tokenizer.tokenize(sentence)
         sentence_field = TextField(tokenized_sentence, self._sentence_token_indexers)
-        agenda = world.get_agenda_for_sentence(sentence, self._add_paths_to_agenda)
+        # TODO(pradeep): Assuming every world gives the same agenda for a sentence. This is true
+        # now, but may change later.
+        agenda = worlds[0].get_agenda_for_sentence(sentence, self._add_paths_to_agenda)
         assert agenda, "No agenda found for sentence: %s" % sentence
         production_rule_fields: List[Field] = []
         instance_action_ids: Dict[str, int] = {}
-        for production_rule in world.all_possible_actions():
+        # TODO(pradeep): Assuming that possible actions are the same in all worlds. This may change
+        # later too.
+        for production_rule in worlds[0].all_possible_actions():
             instance_action_ids[production_rule] = len(instance_action_ids)
             field = ProductionRuleField(production_rule,
                                         terminal_indexers=self._terminal_indexers,
                                         nonterminal_indexers=self._nonterminal_indexers,
-                                        is_nonterminal=lambda x: x not in world.terminal_productions,
+                                        is_nonterminal=lambda x: x not in worlds[0].terminal_productions,
                                         context=tokenized_sentence)
             production_rule_fields.append(field)
         action_field = ListField(production_rule_fields)
         # agenda_field contains indices into actions.
         agenda_field = ListField([IndexField(instance_action_ids[action], action_field)
                                   for action in agenda])
-        world_field = MetadataField(world)
+        worlds_field = ListField([MetadataField(world) for world in worlds])
         fields = {"sentence": sentence_field,
                   "agenda": agenda_field,
-                  "world": world_field,
+                  "worlds": worlds_field,
                   "actions": action_field}
-        if label:
-            label_field = LabelField(label, label_namespace='denotations')
-            fields["label"] = label_field
+        if labels:
+            labels_field = ListField([LabelField(label, label_namespace='denotations') for label in
```

Comment:
It's easier to read if you wrap on the `for`.

Reply:
Done.